### PR TITLE
Zoom to all devices on start

### PR DESCRIPTION
**Pull request by Abyss777**
Zoom to all devices on start if lat/lon/zoom preferences are not set.

We might add some button "Zoom to all" somewhere in future.

### CODE REVIEW

File: web/app/controller/Root.js [hunk 1]
```diff
@@ -167,6 +167,9 @@ Ext.define('Traccar.controller.Root', {
                 self.updateEvents(data.events);
             }
         };
+        if (first) {
+            this.first = true;
```
Comment (Abyss777) [inline web/app/controller/Root.js]: I meant global for `Root.js`

File: web/app/view/MapController.js [hunk 2]
```diff
@@ -98,5 +99,30 @@ Ext.define('Traccar.view.MapController', {
                 return true;
             }, this);
         }
+    },
+
+    zoomToAllDevices: function () {
+        var data, i, point, minx, miny, maxx, maxy;
```
Comment (tananaev) [inline web/app/view/MapController.js]: We have this login somewhere already, so maybe we can share it.

File: web/app/controller/Root.js (hunk 1 above)
Comment (tananaev) [inline web/app/controller/Root.js]: It would be nice if we could use method parameter instead of instance variable.

Comment (Abyss777) [inline web/app/controller/Root.js]: The problem is that `updatePositions` is called from `socket.onmessage` callback and we need switch it off after first frame.

Comment (tananaev) [inline web/app/controller/Root.js]: We can set local variable to false.

**Comment by Abyss777** [comment]
Something like that? (last commit)

File: web/app/controller/Root.js [hunk 3]
```diff
@@ -124,7 +124,7 @@ Ext.define('Traccar.controller.Root', {
     },
 
     asyncUpdate: function (first) {
-        var self = this, protocol, pathname, socket;
+        var self = this, protocol, pathname, socket, firstPositions = first;
```
Comment (tananaev) [inline web/app/controller/Root.js]: I think you can just use `first` directly.

**Comment by Abyss777** [comment]
Removed intermediate variable

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**